mlrun/api/api/endpoints/notifications.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import http
+
+import fastapi
+import fastapi.concurrency
+import sqlalchemy.orm
+
+import mlrun.api.api.deps
+import mlrun.api.crud.notifications
+import mlrun.api.utils.auth.verifier
+import mlrun.api.utils.clients.chief
+import mlrun.api.utils.singletons.project_member
+import mlrun.common.schemas
+from mlrun.utils import logger
+
+router = fastapi.APIRouter(prefix="/projects/{project}/notifications")
+
+
+CHIEF_REDIRECTED_NOTIFICATIONS = [
+    "schedule",
+]
+
+
+@router.put("", status_code=http.HTTPStatus.ACCEPTED.value)
+async def set_object_notifications(
+    project: str,
+    request: fastapi.Request,
+    set_notifications_request: mlrun.common.schemas.SetNotificationRequest = fastapi.Body(
+        ...
+    ),
+    auth_info: mlrun.common.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: sqlalchemy.orm.Session = fastapi.Depends(
+        mlrun.api.api.deps.get_db_session
+    ),
+):
+    await fastapi.concurrency.run_in_threadpool(
+        mlrun.api.utils.singletons.project_member.get_project_member().ensure_project,
+        db_session,
+        project,
+        auth_info=auth_info,
+    )
+
+    # check permission per object type
+    await mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        getattr(
+            mlrun.common.schemas.AuthorizationResourceTypes,
+            set_notifications_request.parent.kind,
+        ),
+        project,
+        resource_name="notifications",
```

Comment:
The `resource_name` should not be `notifications`. It should be the name of the actual resource in question - since you're using the parent kind for authz, then it should be the parent's name